includes/Core/Tags/AMP_Tag.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Tags\AMP_Tag
+ *
+ * @package   Google\Site_Kit\Core\Tags
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Tags;
+
+/**
+ * Base class for AMP tag.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+abstract class AMP_Tag extends Tag implements Blockable_Tag_Interface {
```

Comment:
Let's call these `Module_AMP_Tag` and `Module_Web_Tag` like in the IB and have in them within `Core\Modules` (or `Core\Modules\Tags`), as they are module-specific classes. In theory we could have module-agnostic `Tag` interfaces and classes, but it doesn't satisfy our use-cases and isn't necessary (for example we need to access `$module_slug` like you're doing here), so let's reflect that in their names.

Reply:
Yep, updated.